### ISSUE TITLE
ODS > Active Focus

### ISSUE THREAD

**Issue by B3nz01d** [Orange-OpenSource/Orange-Boosted-Bootstrap#1189]
# Active Focus

## Requirement

**As a** user
**I want** to be able to use a keyboard arrow
**So That** I can select the various items in a page and that they are visually identified

[DSM reference](toBeUpdtaed)

## General description
A selection sould hoghligh all avalable components to understand that they are currently selected

#### Component
* all focus items are using the following color:
  * white background: orange accessible
  * black background: brand orange 
  * other background (using the supporting colors): black
* the focus outline is sitting 2px away from the edge of the texte or the contained item
* the focus line is 3px thikness
* on the drop down the opening arrow would stay in white (not use the black hoover state when selected)
* /!\ on the drop down selected items check if a white separator could be added in) - this could also be applied to the table selected row
* the focus should follow the form of the focus item
* the toggle switch is focusing on the entire item
* on the tag item the focus should be applied to the entire item and not specifically on the delete crosss
* the focus state should not resize the component itself
* the focus state can overlay on top of other items of the componants
* the focus state shouldn't be animated
* the focus state should appear on keyboard focus only

Link to the new wcag 2.2: https://www.w3.org/TR/WCAG22/

### Variants
* Drop down: the selected item should alow to have a white outline between the black selected item and the focus line
    * propose focus inside the list
    * propose focus with the focus outside le list  <- prefered option
* Stepper: is it possible to have the outline follow the exact step form or does it need to be a rectangle outline ???
* Lists: check the selected item gray background and the orange contrast, might need to use the white spacing here as well
* Images for carousel:
  *  carourousel navigation: should use a white outline + an orange focus
  *  pagination dots: the pill holding the dots would switch to full white background and the orange rounded focus is applyed to the size of the dot
* Global header icon with a badge: could the we make the focus go under the badge so that te notification number is still visible???
*  

### States
* NA

### Internationalization
* NA

### Interaction
* All: 
  * The user is able to select a text using the default OS behavior
* Touch: 
  * Nothing Specific
* Keyboard: 
  * Nothing specific

### Animations
* there should not be any animation of the focus with zoom effect - /!\ check if this has an impact from the bootstrap source

### Visual Accessibility
* taken into account within the design creation process

### Usage Accessibility
* (?) Focus order - NA
* (?) Focus loss expected behavior - NA
* (?) Screen Reader - default OS behavior

## Implementation tasks
- [x] #1190
- [ ] #1237

**Comment by MewenLeHo** [comment]
**Question to ODS ? Should the outline be animated such as right now ?**
According to Franco, we have to remove the animation.

**Comment by MewenLeHo** [comment]
Checks&Radios > Switches: we need to find a better solution. The current one is a bit dirty because of the CSS filter on the component (turns the orange focus into a blue one because of the filter: unvert(1)).

**Comment by MewenLeHo** [comment]
Accordion when opened : only solution seems to change accordions-buttons background color to transparent, may bring side effects.

**Comment by louismaximepiton** [comment]
I can't remember why we had a box-shadow around the buttons, If you know, could you remind me ?

**Comment by louismaximepiton** [comment]
Sum up after a spec meeting: 
- Agreement of the assembly to set the `$outline-offset` at `2px`.
- The focus should appear when focused with keyboard only.

**Comment by MewenLeHo** [comment]
We will need to discuss about the back to top button with label, we can't set the focus on the label because it's in a before element which is not in the dom.
